hack/man-page-checker

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env bash
+#
+# man-page-checker - validate and cross-reference man page names
+#
+# This is the script that cross-checks BETWEEN MAN PAGES. It is not the
+# script that cross-checks that each option in skopeo foo --help is listed
+# in skopeo-foo.1.md and vice-versa; that one is xref-helpmsgs-manpages.
+#
+# IMPORTANT NOTE: this script runs on Macs, on which sed is an ancient
+# non-gnu version. To make sed work on all platforms, we invoke with '-E'.
```

Comment:
(I don’t expect a high risk of the inputs or outputs varying across architectures; so if you wanted to make this a separate `make` target that isn’t run on the “build+unit-test on macOS” Cirrus task, that would be perfectly fine with me.

OTOH now that the work has been done, there might be no particular reason to do _even more_ work to drop macOS…)